### PR TITLE
Player Cooldown Bypass Fix

### PR DESCRIPTION
**Pull request by nicholasrobertm**
fix(cooldown) prevent player logging out bypassing cooldown

This was tested and confirmed to work on my test server

### CODE REVIEW

File: forge16/src/main/java/com/envyful/battle/tower/listener/PlayerLogoutListener.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        var attribute = player.getAttribute(BattleTowerAttribute.class);
+
+        if (attribute == null){
+            return;
+        }
+
+        if (attribute.isAttempting()){
+            attribute.finishAttempt();
+        }
```
Comment (danorris709) [inline forge16/src/main/java/com/envyful/battle/tower/listener/PlayerLogoutListener.java]: Please follow the same formatting as the rest of the project

File: forge20/src/main/java/com/envyful/battle/tower/listener/PlayerLogoutListener.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        var attribute = player.getAttribute(BattleTowerAttribute.class);
+
+        if (attribute == null){
+            return;
+        }
+
+        if (attribute.isAttempting()){
+            attribute.finishAttempt();
+        }
+    }
```
Comment (danorris709) [inline forge20/src/main/java/com/envyful/battle/tower/listener/PlayerLogoutListener.java]: Please follow the same formatting as the rest of the project

File: forge16/src/main/java/com/envyful/battle/tower/player/BattleTowerAttribute.java [hunk 3]
```diff
@@ -353,6 +353,10 @@ public void finishAttempt() {
         this.currentFloor = 0;
     }
 
+    public boolean isAttempting() {
+        return this.attemptStart != -1 && currentFloor > 0;
```
Comment (danorris709) [inline forge16/src/main/java/com/envyful/battle/tower/player/BattleTowerAttribute.java:357]: Why is the check for the current floor being greater than 0 important?

File: forge16/src/main/java/com/envyful/battle/tower/listener/PlayerLogoutListener.java (hunk 1 above)
Comment (nicholasrobertm) [reply forge16/src/main/java/com/envyful/battle/tower/listener/PlayerLogoutListener.java]: fixed I think

File: forge16/src/main/java/com/envyful/battle/tower/player/BattleTowerAttribute.java (hunk 3 above)
Comment (nicholasrobertm) [reply forge16/src/main/java/com/envyful/battle/tower/player/BattleTowerAttribute.java:357]: I think the floor being > 0 is probably the actual correct check here, but I did both for sanity after looking at finishAttempt() & startAttempt(). Only having the attempt start had a side effect for some reason of making some players respawn at the battletower every time they logged out, maybe likely due to historical player battletower data. As to why in the DB their floor was set to 0 and the attempt start was not -1, I'm guessing was just a side effect of people disconnecting / crashing themselves in the battletower at some point.

I re-tested this after this change on my test server with an affected users data and its working as expected now

**Pull request merged**